src/bin/lttng-sessiond/Makefile.am

```diff
@@ -59,21 +59,18 @@ lttng_sessiond_SOURCES = utils.c utils.h \
                        action-executor.c action-executor.h\
                        trigger-error-query.c
 
-lttng_sessiond_LDFLAGS = -rdynamic
-
 if HAVE_LIBLTTNG_UST_CTL
-lttng_sessiond_SOURCES += trace-ust.c ust-registry.c ust-app.c \
+liblttng_sessiond_comm_la_SOURCES += trace-ust.c ust-registry.c ust-app.c \
 			ust-consumer.c ust-consumer.h notify-apps.c \
 			ust-metadata.c ust-clock.h agent-thread.c agent-thread.h \
 			ust-field-utils.h ust-field-utils.c \
 			ust-sigbus.c
 endif
 
-# Add main.c at the end for compile order
-lttng_sessiond_SOURCES += lttng-sessiond.h main.c
+liblttng_sessiond_comm_la_LDFLAGS = -no-undefined -static
```

Comment:
I added `-no-undefined` because I was scared of adding undefined references without realizing. I added `-static` as that felt like a good idea for something that is never installed and only used for linking. If you feel its better to remove them I will do that.

Reply:
Yeah, I would remove both of those flag, libtool will handle it.

Reply:
Done.